app/services/weighted_routing_service.rb

```diff
@@ -0,0 +1,32 @@
+class WeightedRoutingService
+  def initialize(collection)
```

Comment:
I moved this to its own file since now we're not always acting on _all_ VitaPartnerState objects from a particular state, nor are they always particularly "real" VitaPartnerState objects.